test/test_linalg.py

```diff
@@ -5597,7 +5596,7 @@ def run_test(dims, eigenvectors, upper):
         for batch_dims, eigenvectors, upper in itertools.product(batch_dims_set, (True, False), (True, False)):
             run_test((5,) + batch_dims, eigenvectors, upper)
 
-    @skipCUDAIfNoMagma
+    @skipCUDAIfNoMagmaAndNoCusolver
```

Comment:
Does this test have the test cases necessary to hit all the paths added in this PR?

Also, historically this test only required MAGMA to run. Is this PR changing PyTorch's behavior so now it needs both MAGMA and cuSolver to run svd? 

Also also, do our test builds include cuSolver to actually run this test?

Reply:
Most linear algebra tests are currently decorated with `skipCUDAIfNoMagma` because they need MAGMA backend. When cusolver backend is added to an operator, we change the decorator to `skipCUDAIfNoMagmaAndNoCusolver`. If will run when magma is available OR cusolver is available, not "AND". 

This test decorator was already enabled for `test_inverse` when its cusolver backend was added in August, and it works fine till now.

Reply:
You're right, it runs if either magma or cusolver is available. The following comment is relevant to this. https://github.com/pytorch/pytorch/pull/48261#issuecomment-756899422